### ISSUE TITLE
Surfshark Isue?

### ISSUE THREAD

**Issue by riddertommie** [guillaumedsde/alpine-qbittorrent-openvpn#13]
After a certain time a get the following error and the container goes to unhealthy.

Thu Jul  9 07:53:56 2020 cd to '/config/openvpn' failed: Permission denied (errno=13)
Thu Jul  9 07:53:56 2020 Exiting due to fatal error

in the container itself I get.

Host not set! Set env 'HEALTH_CHECK_HOST'. For now, using default google.com ping: sendto: Operation not permitted PING google.com (172.217.4.46): 56 data bytes Network is down

putting > - HEALTH_CHECK_HOST=8.8.8.8 in docker-compose and lets wait.

**Comment by riddertommie** [comment]
> After a certain time a get the following error and the container goes to unhealthy.
> 
> Thu Jul 9 07:53:56 2020 cd to '/config/openvpn' failed: Permission denied (errno=13)
> Thu Jul 9 07:53:56 2020 Exiting due to fatal error
> 
> in the container itself I get.
> 
> Host not set! Set env 'HEALTH_CHECK_HOST'. For now, using default google.com ping: sendto: Operation not permitted PING google.com (172.217.4.46): 56 data bytes Network is down
> 
> putting > - HEALTH_CHECK_HOST=8.8.8.8 in docker-compose and lets wait.
> 
> I'm Using
> image: guillaumedsde/alpine-qbittorrent-openvpn:python and did a rebuild with the latest image.

The problem is still there
Fri Jul 10 07:44:17 2020 cd to '/config/openvpn' failed: Permission denied (errno=13)
Fri Jul 10 07:44:17 2020 Exiting due to fatal error

**Comment by guillaumedsde** [comment]
> > After a certain time a get the following error and the container goes to unhealthy.
> > Thu Jul 9 07:53:56 2020 cd to '/config/openvpn' failed: Permission denied (errno=13)
> > Thu Jul 9 07:53:56 2020 Exiting due to fatal error
> > in the container itself I get.
> > Host not set! Set env 'HEALTH_CHECK_HOST'. For now, using default google.com ping: sendto: Operation not permitted PING google.com (172.217.4.46): 56 data bytes Network is down
> > putting > - HEALTH_CHECK_HOST=8.8.8.8 in docker-compose and lets wait.
> > I'm Using
> > image: guillaumedsde/alpine-qbittorrent-openvpn:python and did a rebuild with the latest image.
> 
> The problem is still there
> Fri Jul 10 07:44:17 2020 cd to '/config/openvpn' failed: Permission denied (errno=13)
> Fri Jul 10 07:44:17 2020 Exiting due to fatal error

Hey, can you `ls -al` the `config` directory on your docker host to see what the permissions are for the config directory?

**Comment by riddertommie** [comment]
The /config directory itself is:

drwxrwxr-x    5 1000     1000          4096 Jul 28 13:55 config

the file within are:

drwxrwxr-x    5 1000     1000          4096 Jul 28 13:55 .
drwxr-xr-x    1 root     root          4096 Jul 30 12:53 ..
drwx------    2 openvpn  openvpn       4096 Jul 30 12:53 openvpn
drwx------    5 1000     1000          4096 Jun 26 09:23 qBittorrent
drwxr-xr-x    2 1000     1000          4096 May 18 18:37 self-signed

FYI i saw that the last image was auto pulled (4 hours ago) so that means if something is wrong i can only see it within 24 hours. 

Cheers!

**Comment by guillaumedsde** [comment]
> The /config directory itself is:
> 
> drwxrwxr-x 5 1000 1000 4096 Jul 28 13:55 config
> 
> the file within are:
> 
> drwxrwxr-x 5 1000 1000 4096 Jul 28 13:55 .
> drwxr-xr-x 1 root root 4096 Jul 30 12:53 ..
> drwx------ 2 openvpn openvpn 4096 Jul 30 12:53 openvpn
> drwx------ 5 1000 1000 4096 Jun 26 09:23 qBittorrent
> drwxr-xr-x 2 1000 1000 4096 May 18 18:37 self-signed
> 
> FYI i saw that the last image was auto pulled (4 hours ago) so that means if something is wrong i can only see it within 24 hours.
> 
> Cheers!

yep, I have now tagged all images with their git commit on the dockerhub so now, it is possible to rollback to previous versions.
if you try to `chmod 755 /config/openvpn` and start a newly pulled container again, I think this might fix the issue.

**Comment by riddertommie** [comment]
I Did that in a previous version but as soon as i recreate the container its changed back. same behavior with this version.

To be sure you mean `chmod 755 /config/openvpn` outside the container?

**Comment by guillaumedsde** [comment]
yep, I removed the code that changed the permissions on the directory, now it just changes the config file's owner

**Comment by riddertommie** [comment]
Looks like it's holding. wil keep a eye on it.

Thanks!

**Comment by guillaumedsde** [comment]
> Looks like it's holding. wil keep a eye on it.
> 
> Thanks!

good to hear don't hesitate to post back with any issue :)

**Comment by riddertommie** [comment]
Cheered to early....

Sun Aug  2 15:13:17 2020 cd to '/config/openvpn' failed: Permission denied (errno=13),
Sun Aug  2 15:13:22 2020 Exiting due to fatal error

**Comment by evelez7** [comment]
From some brief experimentation, it seems that my specific problem came from where I set the volume locations. They were in `/srv` in my Debian WSL, so that might have caused some problems. I have them located in `/mnt/c` now and they seem to be ok.
